sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1644 @@
+contract caller allowance for that utility contract.
+
+#### Delegated stackers
+
+Delegation operators wishing to repeatedly stack with a given user
+without cooldown may do the following:
+
+1. Issue a contract-call on behalf of the delegated stacker they wish
+   to re-stack:
+   
+```clarity
+   (contract-call SP000000000000000000002Q6VF78.pox-2 delegate-stack-extend
+     ...)
+```
+
+2. That contract-call *locks but does not commit* the user's funds. In
+   order for the delegation operator to register the reward address for
+   those funds, they must invoke the aggregation commit function (just as
+   when they invoke `delegate-stack-stx`):
+
+```clarity
+   (contract-call SP000000000000000000002Q6VF78.pox-2 stack-aggregation-commit
+     ...)
+```
+
+3. If the delegator wishes to *increase* the amount that the user has
+   stacked, they must separately issue a `delegate-stack-increase` call
+   which locks but does not commit the increased funds.
+
+### Usecase: Stacking with Multiple PoX Addresses
+
+#### Direct stackers
+
+Direct stackers cannot use multiple reward address for the same Stacking
+address. This is because the direct stacking interface does not support
+partial stacking operations (i.e., separating the locking operation from
+the stacking operation), which are necessary for supporting this use case.
+Users who wish to do this should use the delegation interfaces.
+
+#### Delegated stackers
+
+Even in PoX-1, delegation operators are capable of assigning multiple
+reward addresses for funds that they manage: each invocation of
+`delegate-stack-stx` can use a different PoX reward address. However,
+if operators wished to split the same _account's_ lockups across
+multiple addresses, this would not be possible.
+
+In PoX-2, however, delegation operators can use
+`delegate-stack-increase` to achieve this. This method allows the
+delegation operator to set a new PoX address to receive the
+"partially stacked" funds from the increase. Once called, the operator
+can use `stack-aggregation-commit` to commit to each PoX reward
+address separately.
+
+## Clarity
+
+If this SIP is ratified, then at the time of the
+Stacks 2.1 network upgrade, the Clarity smart
+contract language will be expanded with new features and some new
+behaviors for existing native methods. In order to support these
+changes, this SIP proposes to introduce *Clarity versioning for smart
+contracts*. In this scheme, each smart contract will be associated
+with a particular Clarity version. The execution environment will
+track the current version for a given execution (in the
+implementation, this will be via the `ContractContext`), and use that
+to select which features are available, and which native method
+implementations will be used. Clarity 2 contracts can invoke Clarity 1
+contracts (and vice-versa), but particular care will need to be taken
+if a new native keyword is used in the Clarity 1 contract's API.
```

Comment:
The SIP should clarify the criteria for incrementing the Clarity version and disclose that there will be a new version every time a native function are introduced.

Reply:
Sure, can do.

Reply:
Added a paragraph explaining why adding, changing, or removing native Clarity functions and global variables necessitates a new Clarity version (and a breaking change).